plugin-api/src/main/java/org/sonar/api/resources/Language.java

```diff
@@ -60,4 +64,17 @@ public interface Language {
   default boolean publishAllFiles() {
     return true;
   }
+
+  /**
+   * Provide a list of patterns following the {@link org.sonar.api.utils.WildcardPattern} syntax. Source files matching any of those patterns will
+   * be associated to this language.
+   * Pattern are considered relative: <code>["pom.xml"]</code> is equivalent to <code>["&#42;&#42;/pom.xml"]</code> <br/>
+   * The filename extension matching is case-insensitive, so declaring <code>["&#42;&#42;/&#42;Test.java"]</code> will match "FooTest.java" and
+   * "FooTest.JAVA" but <strong>not</strong> "FooTEST.java".<br/>
+   * If both {@link #getFileSuffixes()} and {@link #filenamePatterns()} are provided, both will be considered.
```

Comment:
If I'm not mistaken, an exception is thrown if a file matches 2 languages. Maybe it's worth documenting that behavior?

Reply:
Done, may I let you review again please and check if this is understandable.

Reply:
thanks, LGTM